test/lib/utils/deprecate.js

```diff
@@ -0,0 +1,186 @@
+            hideWarningsStub = new sinon.stub(Logger.prototype, 'hideWarnings');
+        });
+
+        beforeEach(function () {
+            hideWarningsStub.reset();
+        });
+
+        after(function () {
+            hideWarningsStub.restore();
+        });
+
+        it('should put deprecate to initialized state if opts are not passed', function () {
+            deprecate.initialize();
+            deprecate.deprecate(messageInfo);
+
+            expect(logStub).to.be.called;
+        });
+
+        it('should put deprecate to initialized state if opts are passed', function () {
+            var opts = { hideWarnings: true };
+
+            deprecate.initialize(opts);
+            deprecate.deprecate(messageInfo);
+
+            expect(logStub).to.be.called;
+        });
+
+        it('should disable warnings displaying if opts.showWarnings passed as false', function () {
+            var opts = { hideWarnings: true };
+
+            deprecate.initialize(opts);
+
+            expect(hideWarningsStub).to.be.called;
+        });
+
+        it('should print delayed messages after initialization', function () {
+            deprecate.deprecate(messageInfo);
+            expect(logStub).to.be.not.called;
+
+            deprecate.initialize();
+            expect(logStub).to.be.calledOnce;
+        });
+    });
+
+    describe('deprecate', function () {
+        it('should throw error if no deprecated module provided', function () {
+            var brokenMessageInfo = {
+                method: 'testMethod',
+                since: 'v1.0.0'
+            };
+            var func = function () {
+                deprecate.deprecate(brokenMessageInfo);
+            };
+
+            expect(func).to.throw('Missing required field: module');
+        });
+
+        it('should delay message until initialization if deprecate is not initialized', function () {
+            var messageInfo = { module: 'deprecated_module', since: 'v1.0.0' };
+
+            deprecate.deprecate(messageInfo);
+            expect(logStub).to.be.not.called;
+
+            deprecate.initialize();
+            expect(logStub).to.be.calledOnce;
+        });
+
+        describe('initialized deprecate', function () {
+            beforeEach(function () {
+                deprecate.initialize();
+            });
+
+            it('should print message if deprecate was already initialized', function () {
+                var messageInfo = { module: 'deprecated_module', since: 'v1.0.0' };
+
+                deprecate.deprecate(messageInfo);
+
+                expect(logStub).to.be.called;
+            });
+
+            it('should print log action as deprecate', function () {
+                var messageInfo = { module: 'deprecated_module', since: 'v1.0.0' };
+                var expectedAction = 'deprecate';
+
+                deprecate.deprecate(messageInfo);
+
+                expect(logStub).to.be.calledWith(expectedAction, sinon.match.any, sinon.match.any);
```

Comment:
вроде, если не указывать аргументы, то они и не матчатся.
И кстати, нет смысла в отдельных переменных:
```js
deprecate.deprecate({module: 'some-module', since: 'v1.0.0'});
Logger.prototype.logWarningAction.should.be.calledWith('deprecate');
```